### DISCUSSION TITLE
:G commit -am "mymessage"

### DISCUSSION THREAD

**Discussion by smckown** [tpope/vim-fugitive#1875] [Q&A]
When I issue **:Git push**, its output is written outside of a window, with the common "Press Enter or type a command to continue" prompt following.

When I issue **:Git commit -am "message"**, the output is captured into a new vim window.  Is there a variant invocation that would dispatch its output more like **:Git push** does?

**:!git commit -am "message"** is an option, but it **:Git** is more useful because it takes into account the directory path of the file in the active buffer.

**Comment by tpope** [comment]
`:Git commit` does use the "Press ENTER" variant by default. It should only use the window version if you've done something like set `pager.commit` in your Git config.

**Comment by smckown** [comment]
Finally spent a minute to track this down a bit better.

Typing `:Git commit ...` works as described.
Recalling and running a `Git commit ...` from history via `q:` also works correctly.

Recalling/running a `Git commit ...` from history using fzf.vim's `:History:` causes the git command output to be placed in a new vim window.

Perhaps this is best resolved in fzf.vim; I've opened https://github.com/junegunn/fzf.vim/issues/1346.

**Reply by tpope** [reply]
Other things that could cause an output window are `:Git -p commit`, `:Git --pager commit`, and `:Git! commit`. That last one in particular sounds like something that could be called accidentally via either FZF bug or user error.
